#!/usr/bin/env python
PACKAGE = "cv"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#Color Masking parameters
gen.add("hueLowerBound",        int_t,    0, "Hue Lower Bound", 5  ,  0, 180)
gen.add("hueUpperBound",        int_t,    0, "Hue Upper Bound", 25 ,  0, 180)
gen.add("satLowerBound",        int_t,    0, "Sat Lower Bound", 222,  0, 255)
gen.add("satUpperBound",        int_t,    0, "Sat Upper Bound", 255,  0, 255)
gen.add("valLowerBound",        int_t,    0, "Val Lower Bound", 222,  0, 255)
gen.add("valUpperBound",        int_t,    0, "Val Upper Bound", 255,  0, 255)
gen.add("redChannelXFactor",    double_t, 0, "Red channel increase factor"   , 1,1, 3)

#Canny Edge Detection Parameters
gen.add("cannyLowerThreshold" , int_t,    0, "Canny Lower Threshold" , 100           ,  1 , 200)
gen.add("cannyUpperThreshold" , int_t,    0, "Canny Upper Threshold" , 200           ,  100 , 300)


#Hough Lines Parameters
gen.add("houghRho"         ,        int_t,    0, "Distance Resolution" , 1           ,  1 , 10)
gen.add("houghTheta"       ,        double_t, 0, "Angular Resolution"  , 3.1415/180  ,  0 , 10)
gen.add("houghThreshold"   ,        int_t,    0, "Min Intersections"   , 20          ,  0 , 50)
gen.add("houghMinLineLength",        int_t,    0, "Min Pixels in a line", 40          ,  10, 180)
gen.add("houghMaxLineGap"   ,        int_t,    0, "Max Pixel Gap"       , 100         ,  0 , 500)

#Gaussian blurring Parameters
gen.add("blurKernelSize"       ,        int_t,    0, "Gaussian Kernel Size"        , 7           ,  1 , 10)
gen.add("blurSigma"        ,        int_t,    0, "Gaussian Standard Deivation" , 2           ,  1 , 10)

#CV Message Smoothing queue size 
gen.add("smoothQueueSize"  ,        int_t,    0, "Smoothing Queue Size" , 4           ,  1 , 20)

exit(gen.generate(PACKAGE, "LaneDetector", "laneDetectorParams"))
